doc/lsst.pipe.base/command-line-task-retargeting-howto.rst

```diff
@@ -0,0 +1,131 @@
+- `lsst.obs.sdss.selectSdssImages.SelectSdssImagesTask`: an version of the task that selects images for co-addition of SDSS stripe 82 images.
+
+How you retarget a subtask and override its config parameters depends on whether the task is specified as an `lsst.pex.config.ConfigurableField` (the most common case) or as an `lsst.pex.config.RegistryField`.
+Both scenarios are described on this page.
+
+.. _command-line-task-retargeting-howto-show-subtasks:
+
+Viewing configured subtasks
+===========================
+
+To see what subtasks are currently configured to run with a command line task, use the :option:`--show tasks <--show>` argument.
+For example:
+
+.. code-block:: bash
+
+   processCcd.py REPOPATH --output output --show tasks
+
+An example of the printed output is:
+
+.. code-block:: text
+
+   sub-tasks:
+   calibrate: lsst.pipe.tasks.calibrate.CalibrateTask
+   calibrate.applyApCorr: lsst.meas.base.applyApCorr.ApplyApCorrTask
+   calibrate.astromRefObjLoader: lsst.meas.algorithms.loadIndexedReferenceObjects.LoadIndexedReferenceObjectsTask
+   calibrate.astrometry: lsst.meas.astrom.astrometry.AstrometryTask
+   calibrate.astrometry.matcher: lsst.meas.astrom.matchOptimisticB.matchOptimisticBContinued.MatchOptimisticBTask
+   ...
+
+This subtask hierarchy is interpreted as follows:
+
+- The ``calibrate`` subtask of the command line task (``processCcd.py``) configured to use `lsst.pipe.tasks.calibrate.CalibrateTask`.
+- ``calibrate`` (again, implemented by `~lsst.pipe.tasks.calibrate.CalibrateTask`) has a subtask named ``astrometry``, which is currently configured to use the `lsst.meas.astrom.astrometry.AstrometryTask` task.
+- ``calibrate.astrometry` has a subtasked named ``matcher``, which is implemented by `~lsst.meas.astrom.matchOptimisticB.matchOptimisticBContinued.MatchOptimisticBTask`.
+
+Note that if the ``calibrate.astrometry`` task is retargeted to a different task class, the subtask of ``calibrate.astrometry`` *may* change (for example, ``calibrate.astrometry.matcher`` may no longer exist).
+
+
+.. _command-line-task-retargeting-howto-configurablefield:
+
+How to retarget a subtask configured as a ConfigurableField with a configuration file
+=====================================================================================
+
+To retarget a subtask specified as an `lsst.pex.config.ConfigurableField`, you must use a configuration file (specified by :option:`--configfile`, see :ref:`command-line-task-config-howto-configfile`).
+Inside a configuration file, retargeting is done in two steps:
+
+1. Import the Task class you intend to use.
+2. Assign that class to to the subtask configuration using the `~lsst.pex.config.ConfigurableField.retarget` method.
+
+For example, to retarget the subtask named ``configurableSubtask`` with a class ``FooTask``, this configuration file should contain:
+
+.. code-block:: python
+
+   from ... import FooTask
+   config.configurableSubtask.retarget(FooTask)
+
```

Comment:
I'm a little sad that the examples start getting more abstract here and are using "Foo" instead of actual LSST things.

After reading this whole section, it's also not clear to me how to tell whether something I want to retarget is a `ConfigurableField` or a `RegistryField` or what the distinction really means.